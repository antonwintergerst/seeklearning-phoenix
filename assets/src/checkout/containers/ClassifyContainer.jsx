import React from 'react';
import AdSelectionForm from '../components/AdSelectionForm';
import { adTypes, adFeatures } from './ClassifyContainer.data';

const ClassifyContainer = () => (
  <div>
    <h3 className="ui header">Choose your style of ad</h3>
    <div className="ui stacked segments">
      <AdSelectionForm adTypes={adTypes} adFeatures={adFeatures} />
    </div>
  </div>
);

export default ClassifyContainer;
