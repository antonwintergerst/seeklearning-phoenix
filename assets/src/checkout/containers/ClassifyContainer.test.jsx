import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import ClassifyContainer from './ClassifyContainer';

// configure enzyme
configure({ adapter: new Adapter() });

describe('ClassifyContainer', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);

  const initialState = {
    checkout: {
      quote: {
        products: [],
        jobs: [],
      },
      job: null,
    },
  };

  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<ClassifyContainer store={store} />);
  });

  it('renders', () => {
    expect(container.length).toEqual(1);
  });

  it('maps state to props', () => {
    expect(container.prop('output')).toEqual(initialState.output);
  });

  it('renders as expected', () => {
    const tree = ReactTestRenderer.create(container).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
