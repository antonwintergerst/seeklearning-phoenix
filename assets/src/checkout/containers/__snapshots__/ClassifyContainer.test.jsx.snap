// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassifyContainer renders as expected 1`] = `
<div>
  container
</div>
`;
